test/e2e/clusterset_test.go

```diff
@@ -316,12 +316,29 @@ var _ = ginkgo.Describe("Testing ManagedClusterSet", func() {
 				return err
 			}, eventuallyTimeout, eventuallyInterval).ShouldNot(gomega.HaveOccurred())
 		})
-		ginkgo.It("globalClusterSet ns should not be created automatically after delete it", func() {
-			ginkgo.By("globalClusterSet ns should be created after deleted")
+		ginkgo.It("globalClusterSet ns and setbinding should not be created automatically after delete it", func() {
+			ginkgo.By("globalClusterSet ns and setbinding should not be created after deleted")
+
+			//global clusterset binding should not be reconciled.
+			gomega.Eventually(func() error {
+				_, err := clusterClient.ClusterV1beta2().ManagedClusterSetBindings(clustersetutils.GlobalSetNameSpace).Get(context.Background(), clustersetutils.GlobalSetName, metav1.GetOptions{})
+				return err
+			}, eventuallyTimeout, eventuallyInterval).ShouldNot(gomega.HaveOccurred())
+
+			err := clusterClient.ClusterV1beta2().ManagedClusterSetBindings(clustersetutils.GlobalSetNameSpace).Delete(context.Background(), clustersetutils.GlobalSetName, metav1.DeleteOptions{})
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+			gomega.Eventually(func() error {
+				_, err := clusterClient.ClusterV1beta2().ManagedClusterSetBindings(clustersetutils.GlobalSetNameSpace).Get(context.Background(), clustersetutils.GlobalSetName, metav1.GetOptions{})
```

Comment:
I am not sure if you want to check the error return here is a not found error `errors.IsNotFound(err)`?